backend/src/equipment/equipment.controller.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  Body,
+  Controller,
+  Post,
+  Get,
+  Delete,
+  Param,
+  Put,
+} from '@nestjs/common';
+import { ScheduleDTO } from '../schedule/dto/schedule.dto';
+import { EquipmentDTO } from './dto/equipment.dto';
+import { EquipmentService } from './equipment.service';
+
+@Controller(EquipmentController.EQUIPMENT_API_PATH)
+export class EquipmentController {
+  public static readonly EQUIPMENT_API_PATH = '/equipment';
+  public static readonly EQUIPMENT_ID_ROUTE = '/:id';
+  public static readonly EQUIPMENT_CONTROLLER_WITH_SCHED = '/schedule/:id';
+  constructor(private readonly equipmentService: EquipmentService) {}
+
+  @Get()
+  async getAllEquipments(): Promise<EquipmentDTO> {
+    return this.equipmentService.getAllEquipments();
+  }
+  @Post()
+  async addEquipment(@Body() data: EquipmentDTO): Promise<EquipmentDTO> {
+    return this.equipmentService.addEquipment(data);
+  }
+  @Post(EquipmentController.EQUIPMENT_CONTROLLER_WITH_SCHED)
+  async addScheduletoEquipment(
+    @Param('id') id: string,
+    @Body() data: ScheduleDTO
+  ): Promise<EquipmentDTO> {
+    return this.equipmentService.addScheduletoEquipment(id, data);
+  }
+  @Delete(EquipmentController.EQUIPMENT_ID_ROUTE)
+  async deleteEquipment(@Param('id') id: string): Promise<EquipmentDTO> {
+    return this.equipmentService.deleteEquipment(id);
+  }
+  @Delete(EquipmentController.EQUIPMENT_CONTROLLER_WITH_SCHED + '/:scheduleId')
+  async deleteScheduleFromEquipment(
+    @Param('id') id: string,
+    @Param('scheduleId') scheduleId: string
+  ): Promise<EquipmentDTO> {
+    return this.equipmentService.deleteScheduleToEquipment(id, scheduleId);
+  }
+  @Put(EquipmentController.EQUIPMENT_ID_ROUTE)
+  async updateEquipment(
+    @Param('id') id: string,
+    @Body() data: EquipmentDTO
+  ): Promise<EquipmentDTO> {
+    return this.equipmentService.updateEquipment(id, data);
+  }
+  @Put(EquipmentController.EQUIPMENT_CONTROLLER_WITH_SCHED + '/:scheduleId')
+  async editScheduleFromEquipment(
+    @Body() data: ScheduleDTO,
+    @Param('id') id: string,
+    @Param('scheduleId') scheduleId: string
+  ): Promise<EquipmentDTO> {
+    return this.equipmentService.updateScheduleEquipment(id, scheduleId, data);
+  }
```

Comment:
we shouldn't need this endpoint. if the frontend wants to update a schedule, they can do it directly in the `ScheduleController` which should update it for the equipment as well